dietpi/dietpi-config

```diff
@@ -430,32 +430,76 @@ A long (or insufficiently manufactured) cable may required a higher boost settin
 
 		elif (( $G_WHIP_RETURNED_VALUE == 16 )); then
 
-			# Known FP's for setting brightness
-			local afp_current_set_brightness=(
-
-				'/sys/class/backlight/intel_backlight'
-				'/sys/class/backlight/rpi_backlight'
-				'/sys/class/backlight/lcd0'
-				'/sys/class/backlight/backlight'
-			)
-
-			MIN_VALUE=0
-			MAX_VALUE=255
-			local fp_brightness current_brightness
-			for fp_brightness in "${afp_current_set_brightness[@]}"
+			local aFp_Brightness_Available=()
+			local aBrightness_Name=()
+			local aBrightness_Current=()
+			local aBrightness_Max=()
+
+			local iTotalAvailable=0
+			local fp_brightness
+			for fp_brightness in /sys/class/backlight/*
 			do
-				[[ -f $fp_brightness/brightness ]] || continue
-				current_brightness=$(<"$fp_brightness/brightness")
-				[[ -f $fp_brightness/max_brightness ]] && MAX_VALUE=$(<"$fp_brightness/max_brightness")
-				break
+				if [[ -f $fp_brightness/brightness && -f $fp_brightness/max_brightness ]] &&
+					disable_error=1 G_CHECK_VALIDINT "$(<"$fp_brightness/brightness")" &&
+					disable_error=1 G_CHECK_VALIDINT "$(<"$fp_brightness/max_brightness")"
+				then
+					aFp_Brightness_Available+=( "$fp_brightness" )
+					aBrightness_Name+=( "${fp_brightness##*/}" )
+					aBrightness_Current+=( "$(<$fp_brightness/brightness)" )
+					aBrightness_Max+=( "$(<$fp_brightness/max_brightness)" )
+					(( iTotalAvailable++ ))
+				fi
 			done
 
-			if disable_error=1 G_CHECK_VALIDINT "$current_brightness"; then
+			# - Generate Additional Menus
+			if (( iTotalAvailable > 0 ))
+			then
+				G_WHIP_MENU_ARRAY=()
+				local aBrightness_Percent=()
+				local i=0
+				# - Work out current brightness %, add to selection menu
+				for (( i=0; i<$iTotalAvailable; i++ ))
+				do
 
-				G_WHIP_DEFAULT_ITEM=$current_brightness
-				if G_WHIP_INPUTBOX "Please enter a brightness value:\n - Min = $MIN_VALUE | Max = $MAX_VALUE" && G_CHECK_VALIDINT "$G_WHIP_RETURNED_VALUE" "$MIN_VALUE" "$MAX_VALUE"; then
+					aBrightness_Percent+=( "$(echo ${aBrightness_Current[i]} ${aBrightness_Max[i]} | awk '{printf "%.0f\n", $1 / $2 * 100}')" )
```

Comment:
Indeed within command substitutions `$(...)`, everything needs to be quoted again. Otherwise unintended globbing and word splitting can happen within the command substitution and lead to an unexpected result before the parent quotes take effect.
```suggestion
					aBrightness_Percent+=( "$(echo "${aBrightness_Current[i]} ${aBrightness_Max[i]}" | mawk '{printf "%.0f\n", $1 / $2 * 100}')" )
```
I also switched from `awk` to `mawk`, which is much faster and supports all we need.